internal/bootstrap/bootstrap_provider.go

```diff
@@ -393,30 +433,76 @@ func (b *GitProviderBootstrapper) getRepository(ctx context.Context) (gitprovide
 
 	if b.personal {
 		userRef := newUserRef(b.provider.SupportedDomain(), b.owner)
-		return b.provider.UserRepositories().Get(ctx, newUserRepositoryRef(userRef, repoName))
+		rs := newUserRepositoryRef(userRef, repoName)
+		if b.repositorySlug != "" {
+			rs.SetSlug(b.repositorySlug)
+		}
+		return b.provider.UserRepositories().Get(ctx, rs)
 	}
 
 	orgRef := newOrganizationRef(b.provider.SupportedDomain(), b.owner, subOrgs)
-	return b.provider.OrgRepositories().Get(ctx, newOrgRepositoryRef(orgRef, repoName))
+	if b.ownerKey != "" {
+		orgRef.SetKey(b.ownerKey)
+	}
+
+	rs := newOrgRepositoryRef(orgRef, repoName)
+	if b.repositorySlug != "" {
+		rs.SetSlug(b.repositorySlug)
+	}
+
+	return b.provider.OrgRepositories().Get(ctx, rs)
+}
+
+// getStashURL returns the URL to clone a repository for a given transport type.
+func (b *GitProviderBootstrapper) getStashURL(rs gitprovider.RepositoryRef, transport gitprovider.TransportType) string {
+	var identity, repository string
+	if b.ownerKey != "" {
+		identity = b.ownerKey
+	} else {
+		identity = fmt.Sprintf("~%s", rs.GetIdentity())
+	}
+
+	if b.repositorySlug != "" {
+		repository = b.repositorySlug
+	} else {
+		repository = rs.GetRepository()
+	}
+	switch transport {
+	case gitprovider.TransportTypeHTTPS:
+		return fmt.Sprintf("https://%s/scm/%s/%s.git", rs.GetDomain(), identity, repository)
+	case gitprovider.TransportTypeGit:
+		return fmt.Sprintf("git@%s:%s/%s.git", rs.GetDomain(), identity, repository)
+	case gitprovider.TransportTypeSSH:
+		trimmedDomain := rs.GetDomain()
+		trimmedDomain = strings.Replace(trimmedDomain, "https://", "", -1)
+		trimmedDomain = strings.Replace(trimmedDomain, "http://", "", -1)
+		return fmt.Sprintf("ssh://git@%s/%s/%s", trimmedDomain, identity, repository)
+	}
+	return ""
 }
 
 // getCloneURL returns the Git clone URL for the given
 // gitprovider.UserRepository. If the given transport type is
 // gitprovider.TransportTypeSSH and a custom SSH hostname is configured,
 // the hostname of the URL will be modified to this hostname.
 func (b *GitProviderBootstrapper) getCloneURL(repository gitprovider.UserRepository, transport gitprovider.TransportType) (string, error) {
-	u := repository.Repository().GetCloneURL(transport)
+	var url string
+	if string(b.provider.ProviderID()) == string(provider.GitProviderStash) {
+		url = b.getStashURL(repository.Repository(), transport)
+	} else {
+		url = repository.Repository().GetCloneURL(transport)
+	}
 	// TODO(hidde): https://github.com/fluxcd/go-git-providers/issues/55
-	if strings.HasPrefix(u, "https://https://") {
-		u = strings.TrimPrefix(u, "https://")
+	if strings.HasPrefix(url, "https://https://") {
```

Comment:
Can we please fix this in go-git-providers?

Reply:
the related issue is closed. This needs another pr to test if fixed.